tests/17-multiple-monitors.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+source ./helpers.bash
+
+TEST_NET="cilium"
+MON_1_OUTPUT=$(mktemp)
+MON_2_OUTPUT=$(mktemp)
+MON_3_OUTPUT=$(mktemp)
+
+function cleanup {
+    docker network rm ${TEST_NET} 2> /dev/null || true
+}
+
+function lines_expected {
+    expected=$1
+    actual=`wc -l $2 | awk '{ print $1 }'`
+
+    if [ $actual -lt $expected ]; then
+      abort "monitor output lines($actual) in $2 is less than $expected"
+    fi
+}
+
+trap cleanup EXIT
+
+docker network create --ipv6 --subnet ::1/112 --ipam-driver cilium --driver cilium ${TEST_NET} || true
+
+cilium monitor > ${MON_1_OUTPUT} &
+MON_1_PID=$!
+cilium monitor > ${MON_2_OUTPUT} &
+MON_2_PID=$!
+cilium monitor > ${MON_3_OUTPUT} &
+MON_3_PID=$!
+
+if ! ps --pid $MON_1_PID  && ! ps --pid $MON_2_PID  && ! ps --pid $MON_3_PID; then
+  abort "expected three monitors to be running"
+fi
+
+docker run -d --net cilium --name demo1 -l client tgraf/netperf
+docker exec -ti demo1 ip -6 address list
+docker exec -ti demo1 ip -6 route list dev cilium0
+cilium endpoint list
+docker rm -f demo1
+
+lines_expected 3 ${MON_1_OUTPUT}
+lines_expected 3 ${MON_2_OUTPUT}
+lines_expected 3 ${MON_3_OUTPUT}
+
+kill ${MON_1_PID} ${MON_2_PID} ${MON_3_PID}
+
+if ! diff3 ${MON_1_OUTPUT} ${MON_2_OUTPUT} ${MON_3_OUTPUT}; then
+  #vimdiff ${MON_1_OUTPUT} ${MON_2_OUTPUT} ${MON_3_OUTPUT}
```

Comment:
Actually I was considering leaving this in because it was so useful, but will remove. Thanks.